# Proyecto RappiPlus: de datos a decisiones de negocio

**Introducción**


El objetivo de este proyecto es evaluar el desempeño del servicio **RappiPlus** para apoyar **decisiones de negocio basadas en datos**.

Se trabajan con múltiples datasets del negocio:

- **rappiplus_orders_raw.csv** → información de pedidos, precios, descuentos y revenue  
- **rappiplus_catalog.csv** → costos de productos, categorías y proveedores  
- **rappiplus_marketing_spend.csv** → inversión en marketing por canal y país  
- **events / users / user_activity (SQL)** → comportamiento del usuario dentro de la plataforma  
- **experiment_checkout_ui.csv** → resultados de un experimento A/B en el checkout  

El análisis sigue una lógica clara y progresiva:

1. 🔍 Evaluar si podemos confiar en los datos (calidad de datos en Python) 

2. 💰 Analizar si el negocio es rentable (revenue, costos y profit)  

3. 🛒 Entender dónde se pierden los usuarios (funnel de conversión)  

4. 🔁 Evaluar si los usuarios regresan (retención por cohortes)  

5. 🧪 Validar si los cambios generan impacto (test estadístico)  

6. 📊 Comunicar los resultados (dashboard en BI)  

A lo largo del proyecto, se transforman datos en insights para responder preguntas clave del negocio y proponer **recomendaciones accionables**.

---

## 🔹 Paso 1: Cargar y validar la calidad de los datos

---

### 1.1 Carga de datos y vista rápida

**🎯 Objetivo:** Familiarizarte con la estructura de los datasets del negocio antes de analizarlos.

**Instrucciones:**

- Importa las librerías necesarias
- Carga los archivos:
  - `rappiplus_orders_raw.csv`
  - `rappiplus_catalog.csv`
  - `rappiplus_marketing_spend.csv`
- Guarda los DataFrames en:
  - `orders`, `catalog`, `marketing`
- Explora cada dataset.

---

In [2]:
# importar librerías
import pandas as pd

In [3]:
# cargar archivos
orders = pd.read_csv('https://practicum-content.s3.amazonaws.com/datasets/rappiplus_orders_raw.csv')
catalog = pd.read_csv('https://practicum-content.s3.amazonaws.com/datasets/rappiplus_catalog.csv')
marketing = pd.read_csv ('https://practicum-content.s3.amazonaws.com/datasets/rappiplus_marketing_spend.csv')

In [4]:
# explorar datasets
orders.info()
display(orders.head())
catalog.info()
display(catalog.head())
marketing.info()
display(marketing.head())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 25100 entries, 0 to 25099
Data columns (total 12 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   id_pedido           25100 non-null  object 
 1   id_usuario          25100 non-null  object 
 2   fecha_hora_pedido   25100 non-null  object 
 3   pais                24800 non-null  object 
 4   dispositivo         25080 non-null  object 
 5   fuente_referencia   25070 non-null  object 
 6   nombre_producto     25070 non-null  object 
 7   categoria_producto  25020 non-null  object 
 8   cantidad            25050 non-null  float64
 9   precio_unitario     25050 non-null  float64
 10  monto_descuento     25050 non-null  float64
 11  monto_total         25100 non-null  float64
dtypes: float64(4), object(8)
memory usage: 2.3+ MB


,id_pedido,id_usuario,fecha_hora_pedido,pais,dispositivo,fuente_referencia,nombre_producto,categoria_producto,cantidad,precio_unitario,monto_descuento,monto_total
0,order_0,user_6993,2025-05-22,Argentina,desktop,organic,Jacket-Winter-M,Moda,2.0,332.69,0.0,665.37
1,order_1,user_1329,2025-06-15,Mexico,desktop,paid_search,Tablet-Standard-64GB,Electronica,1.0,176.86,5.0,171.86
2,order_2,user_3194,2025-05-02,Argentina,desktop,social,Blender-XL-Red,Hogar,2.0,102.99,10.0,195.99
3,order_3,user_4510,2025-06-09,Colombia,mobile,social,Tablet-Standard-64GB,Electronica,1.0,257.87,15.0,242.87
4,order_4,user_5044,2025-03-30,Argentina,desktop,paid_search,Blender-XL-Red,Hogar,1.0,336.28,0.0,336.28


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7 entries, 0 to 6
Data columns (total 4 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   nombre_producto     7 non-null      object 
 1   categoria_producto  7 non-null      object 
 2   costo_unitario      7 non-null      float64
 3   proveedor           7 non-null      object 
dtypes: float64(1), object(3)
memory usage: 352.0+ bytes


,nombre_producto,categoria_producto,costo_unitario,proveedor
0,Laptop-Gaming-16GB,Electrónica,280.68,"Fuller, Pena and Myers"
1,Phone-Pro-128GB,Electrónica,10.12,King Ltd
2,Tablet-Standard-64GB,Electrónica,25.21,Bowers LLC
3,Blender-XL-Red,Hogar,176.64,Long-Reid
4,Vacuum-Pro-Black,Hogar,16.60,"Rivera, Carr and Finley"


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1620 entries, 0 to 1619
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   fecha       1620 non-null   object 
 1   pais        1620 non-null   object 
 2   id_campaña  1620 non-null   object 
 3   canal       1519 non-null   object 
 4   gasto       1620 non-null   float64
dtypes: float64(1), object(4)
memory usage: 63.4+ KB


,fecha,pais,id_campaña,canal,gasto
0,2025-01-01,Mexico,organic_Mexico,organic,2446.25
1,2025-01-01,Mexico,paid_search_Mexico,paid_search,2704.34
2,2025-01-01,Mexico,social_Mexico,social,2045.01
3,2025-01-01,Colombia,organic_Colombia,organic,2597.21
4,2025-01-01,Colombia,paid_search_Colombia,paid_search,1771.40


In [5]:
# revisión de dimensiones
print("Orders:", orders.shape)
print("Catalog:", catalog.shape)
print("Marketing:", marketing.shape)

Orders: (25100, 12)
Catalog: (7, 4)
Marketing: (1620, 5)


In [6]:
# revisión valores ausentes
display(orders.isna().sum())
display(catalog.isna().sum())
display(marketing.isna().sum())

id_pedido               0
id_usuario              0
fecha_hora_pedido       0
pais                  300
dispositivo            20
fuente_referencia      30
nombre_producto        30
categoria_producto     80
cantidad               50
precio_unitario        50
monto_descuento        50
monto_total             0
dtype: int64

nombre_producto       0
categoria_producto    0
costo_unitario        0
proveedor             0
dtype: int64

fecha           0
pais            0
id_campaña      0
canal         101
gasto           0
dtype: int64

In [7]:
# revisión duplicados completos
print("Duplicados en orders:", orders.duplicated().sum())
print("Duplicados en catalog:", catalog.duplicated().sum())
print("Duplicados en marketing:", marketing.duplicated().sum())

Duplicados en orders: 100
Duplicados en catalog: 0
Duplicados en marketing: 0


In [8]:
# revisión categorías
display(orders["pais"].value_counts(dropna=False))
display(orders["dispositivo"].value_counts(dropna=False))
display(orders["fuente_referencia"].value_counts(dropna=False))
display(orders["categoria_producto"].value_counts(dropna=False))
display(marketing["canal"].value_counts(dropna=False))
display(marketing["pais"].value_counts(dropna=False))

Colombia     7520
Mexico       7502
Argentina    7291
mexico        865
colombia      823
argentina     799
NaN           300
Name: pais, dtype: int64

desktop    12759
mobile     12321
NaN           20
Name: dispositivo, dtype: int64

social         8428
organic        8329
paid_search    8313
NaN              30
Name: fuente_referencia, dtype: int64

Hogar          8385
Moda           8323
Electronica    8312
NaN              80
Name: categoria_producto, dtype: int64

paid_search    507
social         506
organic        506
NaN            101
Name: canal, dtype: int64

Argentina    540
Colombia     540
Mexico       540
Name: pais, dtype: int64

In [9]:
#revisión duplicados dataset orders
duplicados_orders = orders[orders.duplicated(keep=False)]
display(duplicados_orders.sort_values("id_pedido").head(10))
print("ID de pedido duplicados:", orders["id_pedido"].duplicated().sum())
print("ID de pedido únicos:", orders["id_pedido"].nunique())

,id_pedido,id_usuario,fecha_hora_pedido,pais,dispositivo,fuente_referencia,nombre_producto,categoria_producto,cantidad,precio_unitario,monto_descuento,monto_total
25023,order_10082,user_690,2025-02-17,Argentina,desktop,social,Sneakers-Urban-42,Moda,2.0,221.34,0.0,442.67
10082,order_10082,user_690,2025-02-17,Argentina,desktop,social,Sneakers-Urban-42,Moda,2.0,221.34,0.0,442.67
25037,order_10709,user_6783,2025-02-16,Colombia,mobile,organic,Jacket-Winter-M,Moda,1.0,170.10,10.0,160.10
10709,order_10709,user_6783,2025-02-16,Colombia,mobile,organic,Jacket-Winter-M,Moda,1.0,170.10,10.0,160.10
25065,order_10829,user_7697,2025-01-23,Argentina,mobile,social,Phone-Pro-128GB,Electronica,2.0,115.54,0.0,231.08
10829,order_10829,user_7697,2025-01-23,Argentina,mobile,social,Phone-Pro-128GB,Electronica,2.0,115.54,0.0,231.08
10856,order_10856,user_1899,2025-03-18,Colombia,desktop,social,Laptop-Gaming-16GB,Electronica,1.0,72.53,0.0,72.53
25044,order_10856,user_1899,2025-03-18,Colombia,desktop,social,Laptop-Gaming-16GB,Electronica,1.0,72.53,0.0,72.53
11449,order_11449,user_1480,2025-03-21,Mexico,desktop,paid_search,Vacuum-Pro-Black,Hogar,1.0,60.55,5.0,55.55
25083,order_11449,user_1480,2025-03-21,Mexico,desktop,paid_search,Vacuum-Pro-Black,Hogar,1.0,60.55,5.0,55.55


ID de pedido duplicados: 100
ID de pedido únicos: 25000


#### Decisión duplicados dataset orders
- Se identificaron 100 registros duplicados en orders. La tabla contiene 25,100 filas, pero únicamente 25,000 identificadores de pedido únicos. Al inspeccionar los registros, se comprobó que las filas repetidas coinciden en todas las columnas. Por ello, deben eliminarse las copias y conservarse 25,000 pedidos únicos.

---

### Revisión y calidad de datos

**🎯 Objetivo:** Detectar y corregir problemas en los datos que puedan afectar el análisis de revenue, costos y rentabilidad.

Se revisan los 3 datasets
- Validar y convertir fechas al formato correcto  
- Revisar variables numéricas (sin negativos o ceros inválidos)  
- Verificar consistencia de montos  
- Eliminar duplicados  
- Revisar variables categóricas 

---

In [10]:
# validación y conversión de fechas
orders["fecha_hora_pedido"] = pd.to_datetime (
    orders["fecha_hora_pedido"],
    errors="coerce"
)

marketing["fecha"] = pd.to_datetime (
    marketing["fecha"],
    errors="coerce"
)
print(orders["fecha_hora_pedido"].dtype)
print(marketing["fecha"].dtype)

print("Fechas inválidas en orders:",
      orders["fecha_hora_pedido"].isna().sum())

print("Fechas inválidas en marketing:",
      marketing["fecha"].isna().sum())

datetime64[ns]
datetime64[ns]
Fechas inválidas en orders: 0
Fechas inválidas en marketing: 0


- Las columnas fecha_hora_pedido y fecha inicialmente estaban almacenadas como object. Se convirtieron al tipo datetime utilizando to_datetime. Después de la conversión se detectaron  0 fechas inválidas en pedidos y 0 en marketing.

In [11]:
# estandarización categoria "pais" dataset orders
orders["pais"] = orders["pais"].str.strip().str.title()
display(orders["pais"].value_counts(dropna=False))

Mexico       8367
Colombia     8343
Argentina    8090
NaN           300
Name: pais, dtype: int64

- La columna pais del dataset orders presentaba categorías duplicadas debido al uso inconsistente de mayúsculas y minúsculas. Se eliminaron posibles espacios y se estandarizaron los nombres, conservando las categorías Mexico, Colombia y Argentina. Después de la estandarización permanecen 300 valores ausentes, cuyo tratamiento se evaluará posteriormente.

In [12]:
#revisión variables númericos 
numericas_orders = [
    "cantidad",
    "precio_unitario",
    "monto_descuento",
    "monto_total"
]
numericas_catalog = ["costo_unitario"]
numericas_marketing = ["gasto"]

display(orders[numericas_orders].describe())
display(catalog[numericas_catalog].describe())
display(marketing[numericas_marketing].describe())

,cantidad,precio_unitario,monto_descuento,monto_total
count,25050.000000,25050.000000,25050.000000,2.510000e+04
mean,7.092735,259.305549,4.500798,2.072680e+03
std,296.277003,138.726461,5.223010,9.894995e+04
min,-2.000000,20.030000,0.000000,-4.926500e+02
25%,1.000000,138.377500,0.000000,1.805075e+02
50%,2.000000,258.715000,0.000000,3.417500e+02
75%,2.000000,380.332500,10.000000,5.185800e+02
max,20000.000000,499.960000,15.000000,8.840200e+06


,costo_unitario
count,7.000000
mean,102.252857
std,111.011563
min,10.120000
25%,16.905000
50%,25.210000
75%,182.975000
max,280.680000


,gasto
count,1620.00000
mean,1772.74292
std,734.43294
min,501.11000
25%,1128.03000
50%,1782.42500
75%,2420.68500
max,2999.36000


In [13]:
#revisión negativos variables númericas
print("Negativos en orders:")
display((orders[numericas_orders] < 0).sum())

print("Negativos en catalog:")
display((catalog[numericas_catalog] < 0).sum())

print("Negativos en marketing:")
display((marketing[numericas_marketing] < 0).sum())

Negativos en orders:


cantidad           4
precio_unitario    0
monto_descuento    0
monto_total        4
dtype: int64

Negativos en catalog:


costo_unitario    0
dtype: int64

Negativos en marketing:


gasto    0
dtype: int64

In [14]:
#revisión cero en variables númericas 
print("Ceros en orders:")
display((orders[numericas_orders] == 0).sum())

print("Ceros en catalog:")
display((catalog[numericas_catalog] == 0).sum())

print("Ceros en marketing:")
display((marketing[numericas_marketing] == 0).sum())

Ceros en orders:


cantidad               0
precio_unitario        0
monto_descuento    12551
monto_total            0
dtype: int64

Ceros en catalog:


costo_unitario    0
dtype: int64

Ceros en marketing:


gasto    0
dtype: int64

In [15]:
print("Cantidades disponibles:")
display(orders["cantidad"].value_counts(dropna=False).sort_index())

print("Casos con cantidad mayor que 10:")
print((orders["cantidad"] > 10).sum())

Cantidades disponibles:


-2.0            1
-1.0            3
 1.0        12394
 2.0        12642
 10000.0        6
 20000.0        4
 NaN           50
Name: cantidad, dtype: int64

Casos con cantidad mayor que 10:
10


- Se identificaron registros numéricos problemáticos: cuatro pedidos con cantidades negativas, diez con cantidades extremas de 10,000 o 20,000 unidades y cincuenta con ausencia simultánea de variables esenciales. Se comprobó que estos grupos representan aproximadamente el 0.25% del dataset. Debido a que no existe información suficiente para corregirlos o reconstruirlos sin introducir supuestos, se decidió excluirlos para evitar distorsiones en los cálculos de ingresos, costos y rentabilidad.

In [16]:
#condición para eliminación registro df orders
condicion_problematicos = (
    (orders["cantidad"] < 0) |
    (orders["cantidad"] > 10) |
    (orders[["cantidad", "precio_unitario", "monto_descuento"]]
        .isna()
        .any(axis=1))
)

print("Registros problemáticos:",
      orders[condicion_problematicos].shape[0])

print(
    "Porcentaje:",
    orders[condicion_problematicos].shape[0] / orders.shape[0] * 100
)


Registros problemáticos: 64
Porcentaje: 0.2549800796812749


In [17]:
#Eliminación registros problematicos df orders
orders = orders[~condicion_problematicos].reset_index(drop=True)

In [18]:
#Validación eliminación registros problematicos 
print("Dimensión posterior:", orders.shape)
print("Negativos:", (orders[numericas_orders] < 0).sum())
print("Ausentes:", orders[numericas_orders].isna().sum())
print("Cantidades disponibles:")
display(orders["cantidad"].value_counts(dropna=False).sort_index())

Dimensión posterior: (25036, 12)
Negativos: cantidad           0
precio_unitario    0
monto_descuento    0
monto_total        0
dtype: int64
Ausentes: cantidad           0
precio_unitario    0
monto_descuento    0
monto_total        0
dtype: int64
Cantidades disponibles:


1.0    12394
2.0    12642
Name: cantidad, dtype: int64

In [19]:
#verificación consistencia de montos
orders["monto_esperado"] = (
    orders["cantidad"] * orders["precio_unitario"]
    - orders["monto_descuento"]
).round(2)
orders["diferencia_monto"] = (
    orders["monto_total"] - orders["monto_esperado"]
).abs().round(2)
display(
    orders["diferencia_monto"]
    .value_counts()
    .sort_index()
)
print("Diferencia mínima:",
      orders["diferencia_monto"].min())

print("Diferencia máxima:",
      orders["diferencia_monto"].max())
montos_inconsistentes = orders[
    orders["diferencia_monto"] > 0.01
]

print("Montos inconsistentes:",
      montos_inconsistentes.shape[0])

0.00    18780
0.01     6256
Name: diferencia_monto, dtype: int64

Diferencia mínima: 0.0
Diferencia máxima: 0.01
Montos inconsistentes: 0


- Se calculó el monto esperado mediante cantidad × precio_unitario − monto_descuento y se comparó con monto_total. Se estableció una tolerancia máxima de $0.01 por posibles efectos de redondeo. De los 25,036 registros analizados, 18,780 coincidieron exactamente y 6,256 presentaron una diferencia de un centavo. No se encontraron montos inconsistentes superiores a la tolerancia, por lo que se conservó el valor original de monto_total.

In [20]:
#eliminación columnas auxiliares "monto_esperado" y "diferencia_monto"
orders = orders.drop(
    columns=["monto_esperado", "diferencia_monto"]
)

In [21]:
# verificación duplicados
print("Duplicados completos:",
      orders.duplicated().sum())
print("ID de pedido duplicados:",
      orders["id_pedido"].duplicated().sum())
print("ID únicos:",
      orders["id_pedido"].nunique())
print("Duplicados en catalog:",
      catalog.duplicated().sum())

print("Duplicados en marketing:",
      marketing.duplicated().sum())


Duplicados completos: 100
ID de pedido duplicados: 100
ID únicos: 24936
Duplicados en catalog: 0
Duplicados en marketing: 0


In [22]:
#eliminación resgistros duplicados orders
orders = orders.drop_duplicates(
    keep="first"
).reset_index(drop=True)

- Se identificaron 100 filas completamente duplicadas en orders. La cantidad de duplicados coincidió con la diferencia entre el número de filas y los identificadores únicos, confirmando que se trataba de copias de pedidos existentes. Se conservó la primera aparición de cada pedido y se eliminaron las repeticiones para evitar sobreestimar ingresos, unidades vendidas y número de pedidos. Los datasets catalog y marketing no presentaron duplicados.

In [23]:
#validación eliminación duplicados
print("Dimensión después de eliminar duplicados:",
      orders.shape)

print("Duplicados completos:",
      orders.duplicated().sum())

print("ID duplicados:",
      orders["id_pedido"].duplicated().sum())

print("ID únicos:",
      orders["id_pedido"].nunique())

Dimensión después de eliminar duplicados: (24936, 12)
Duplicados completos: 0
ID duplicados: 0
ID únicos: 24936


In [24]:
# Columnas categóricas

categoricas_orders = [
    "pais",
    "dispositivo",
    "fuente_referencia",
    "nombre_producto",
    "categoria_producto"
]

categoricas_catalog = [
    "nombre_producto",
    "categoria_producto",
    "proveedor"
]

categoricas_marketing = [
    "pais",
    "id_campaña",
    "canal"
]

# Revisión de categorías en orders
for columna in categoricas_orders:
    print(f"\n--- orders: {columna} ---")
    display(orders[columna].value_counts(dropna=False))
    
# Revisión de categorías en catalog
for columna in categoricas_catalog:
    print(f"\n--- catalog: {columna} ---")
    display(catalog[columna].value_counts(dropna=False))

# Revisión de categorías en marketing
for columna in categoricas_marketing:
    print(f"\n--- marketing: {columna} ---")
    display(marketing[columna].value_counts(dropna=False))


--- orders: pais ---


Mexico       8322
Colombia     8287
Argentina    8031
NaN           296
Name: pais, dtype: int64


--- orders: dispositivo ---


desktop    12676
mobile     12240
NaN           20
Name: dispositivo, dtype: int64


--- orders: fuente_referencia ---


social         8382
organic        8267
paid_search    8257
NaN              30
Name: fuente_referencia, dtype: int64


--- orders: nombre_producto ---


Blender-XL-Red          4176
Vacuum-Pro-Black        4170
Jacket-Winter-M         4166
Sneakers-Urban-42       4129
Laptop-Gaming-16GB      2768
Tablet-Standard-64GB    2764
Phone-Pro-128GB         2733
NaN                       30
Name: nombre_producto, dtype: int64


--- orders: categoria_producto ---


Hogar          8346
Moda           8295
Electronica    8265
NaN              30
Name: categoria_producto, dtype: int64


--- catalog: nombre_producto ---


Tablet-Standard-64GB    1
Phone-Pro-128GB         1
Blender-XL-Red          1
Jacket-Winter-M         1
Laptop-Gaming-16GB      1
Sneakers-Urban-42       1
Vacuum-Pro-Black        1
Name: nombre_producto, dtype: int64


--- catalog: categoria_producto ---


Electrónica    3
Moda           2
Hogar          2
Name: categoria_producto, dtype: int64


--- catalog: proveedor ---


Rivera, Carr and Finley    1
Fuller, Pena and Myers     1
Greene-Smith               1
Mcmillan-Rhodes            1
Bowers LLC                 1
King Ltd                   1
Long-Reid                  1
Name: proveedor, dtype: int64


--- marketing: pais ---


Argentina    540
Colombia     540
Mexico       540
Name: pais, dtype: int64


--- marketing: id_campaña ---


paid_search_Colombia     180
organic_Colombia         180
paid_search_Argentina    180
social_Argentina         180
organic_Argentina        180
social_Colombia          180
paid_search_Mexico       180
organic_Mexico           180
social_Mexico            180
Name: id_campaña, dtype: int64


--- marketing: canal ---


paid_search    507
social         506
organic        506
NaN            101
Name: canal, dtype: int64

In [25]:
# Unificar categoría entre orders y catalog


orders["categoria_producto"] = (
    orders["categoria_producto"]
    .str.strip()
    .replace({"Electrónica": "Electronica"})
)

catalog ["categoria_producto"] = (
    catalog["categoria_producto"]
    .str.strip()
    .replace({"Electrónica": "Electronica"})
)



# Eliminar pedidos sin producto y categoría

orders = orders.dropna(
    subset=["nombre_producto", "categoria_producto"]
).reset_index(drop=True)

# Conservar otros ausentes como categoría desconocida

orders["pais"] = orders["pais"].fillna("Desconocido")

orders["dispositivo"] = (
    orders["dispositivo"].fillna("desconocido")
)

orders["fuente_referencia"] = (
    orders["fuente_referencia"].fillna("desconocido")
)

# Recuperar canal de marketing desde id_campaña

canal_desde_campaña = (
    marketing["id_campaña"]
    .str.rsplit("_", n=1)
    .str[0]
)

marketing["canal"] = marketing["canal"].fillna(
    canal_desde_campaña
)

In [26]:
# validación correcciones varoiables categoricas 
print("Dimensión final de orders:", orders.shape)

print("\nAusentes en orders:")
display(orders[categoricas_orders].isna().sum())

print("\nAusentes en catalog:")
display(catalog[categoricas_catalog].isna().sum())

print("\nAusentes en marketing:")
display(marketing[categoricas_marketing].isna().sum())

print("\nCategorías finales de producto:")
display(
    orders["categoria_producto"].value_counts(dropna=False)
)

print("\nCanales finales de marketing:")
display(
    marketing["canal"].value_counts(dropna=False)
)

Dimensión final de orders: (24906, 12)

Ausentes en orders:


pais                  0
dispositivo           0
fuente_referencia     0
nombre_producto       0
categoria_producto    0
dtype: int64


Ausentes en catalog:


nombre_producto       0
categoria_producto    0
proveedor             0
dtype: int64


Ausentes en marketing:


pais          0
id_campaña    0
canal         0
dtype: int64


Categorías finales de producto:


Hogar          8346
Moda           8295
Electronica    8265
Name: categoria_producto, dtype: int64


Canales finales de marketing:


social         540
organic        540
paid_search    540
Name: canal, dtype: int64

- Se revisaron las variables categóricas mediante value_counts(dropna=False) para detectar diferencias de escritura, espacios, categorías inesperadas y valores ausentes. Se estandarizó la categoría Electronica como Electrónica para mantener consistencia entre orders y catalog. Los 30 pedidos sin producto ni categoría fueron eliminados porque no podían vincularse con el catálogo para calcular costos y rentabilidad. Los valores ausentes de país, dispositivo y fuente de referencia se conservaron bajo la categoría Desconocido. En marketing, los 101 canales ausentes se recuperaron a partir de id_campaña. Después de la limpieza, no quedaron valores categóricos ausentes.

In [27]:
#validación final 
print("Dimensiones finales:")
print("Orders:", orders.shape)
print("Catalog:", catalog.shape)
print("Marketing:", marketing.shape)

print("\nAusentes totales:")
print("Orders:", orders.isna().sum().sum())
print("Catalog:", catalog.isna().sum().sum())
print("Marketing:", marketing.isna().sum().sum())

print("\nDuplicados:")
print("Orders:", orders.duplicated().sum())
print("Catalog:", catalog.duplicated().sum())
print("Marketing:", marketing.duplicated().sum())

print("\nTipos de datos:")
orders.info()
catalog.info()
marketing.info()

Dimensiones finales:
Orders: (24906, 12)
Catalog: (7, 4)
Marketing: (1620, 5)

Ausentes totales:
Orders: 0
Catalog: 0
Marketing: 0

Duplicados:
Orders: 0
Catalog: 0
Marketing: 0

Tipos de datos:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 24906 entries, 0 to 24905
Data columns (total 12 columns):
 #   Column              Non-Null Count  Dtype         
---  ------              --------------  -----         
 0   id_pedido           24906 non-null  object        
 1   id_usuario          24906 non-null  object        
 2   fecha_hora_pedido   24906 non-null  datetime64[ns]
 3   pais                24906 non-null  object        
 4   dispositivo         24906 non-null  object        
 5   fuente_referencia   24906 non-null  object        
 6   nombre_producto     24906 non-null  object        
 7   categoria_producto  24906 non-null  object        
 8   cantidad            24906 non-null  float64       
 9   precio_unitario     24906 non-null  float64       
 10  monto_descuento    

---
**📦 Exportación**: Una vez finalizada la limpieza, se exportan los datasets para utilizarlos en la última etapa del proyecto.

In [28]:
# exportar datasets
orders.to_csv('orders_clean.csv', index=False)
catalog.to_csv('catalog_clean.csv', index=False)
marketing.to_csv('marketing_clean.csv', index=False)

---

## 🔹 Paso 2: Analizar si el negocio es rentable

### 2.1 Cálculo de KPIs principales

**🎯 Objetivo:** Calcular los indicadores clave del negocio para evaluar ingresos, costos y rentabilidad.

Se usan los 3 datasets (`orders`, `catalog`, `marketing_spend`):

**📊 Parte 1: Rentabilidad del negocio**
- ¿Cuál es el ingreso total (revenue)? 
- ¿Cuál es el costo total? 
- ¿Cuánto se ha invertido en marketing? 
- ¿El negocio es rentable? (calcular profit)  

---

**📈 Parte 2: Comportamiento de ventas**
- ¿Cuál es el ticket promedio por orden? 
- ¿Cuál es la cantidad promedio de productos por orden? 
- ¿Cuál es el producto más vendido?
- ¿Cuánto se ha gastado en marketing por canal? 

In [29]:
#KPIs Rentabilidad del negocio 
#Calculo de ingreso total 
revenue_total = orders["monto_total"].sum()
print(f"Revenue total: ${revenue_total:,.2f}")

# Incorporación el costo unitario del catálogo
ventas = orders.merge(catalog[["nombre_producto", "costo_unitario"]],on="nombre_producto",how="left")

#Validación del merge 
print("Filas originales:", orders.shape[0])
print("Filas después del merge:", ventas.shape[0])
print("Costos ausentes:", ventas["costo_unitario"].isna().sum())

# Calculo el costo total
costo_total = ( ventas["cantidad"] * ventas["costo_unitario"]).sum()
print(f"Costo total: ${costo_total:,.2f}")

#Calculo inversión total en marketing
gasto_marketing = marketing["gasto"].sum()
print(f"Inversión total en marketing: "f"${gasto_marketing:,.2f}")

#Calculo Profit(rentabilidad)
profit = revenue_total - costo_total - gasto_marketing
print(f"Profit: ${profit:,.2f}")

Revenue total: $9,610,018.94
Filas originales: 24906
Filas después del merge: 24906
Costos ausentes: 0
Costo total: $3,828,869.01
Inversión total en marketing: $2,871,843.53
Profit: $2,909,306.40


- Durante el periodo analizado, el negocio generó un revenue total de 9,610,018.94. El costo de los productos vendidos fue de 3,828,869.01 y la inversión en marketing ascendió a 2,871,843.53. Después de descontar ambos conceptos, se obtuvo un profit positivo de 2,909,306.40. Por lo tanto, el negocio fue rentable durante el periodo analizado.

In [30]:
#KPIs Comportamiento de ventas
#Calculo ticket promedio 
ticket_promedio = (revenue_total/ orders["id_pedido"].nunique())
print(f"Ticket promedio: ${ticket_promedio:,.2f}")

#Calculo cantidad promedio de producto por orden 
productos_promedio = (orders.groupby("id_pedido")["cantidad"].sum().mean())
print("Cantidad promedio de productos por orden: "f"{productos_promedio:.2f}")


#Calculo producto más vendido 
# Unidades vendidas por producto
ventas_por_producto = (orders.groupby("nombre_producto")["cantidad"].sum().sort_values(ascending=False))
# Producto más vendido
producto_mas_vendido = ventas_por_producto.idxmax()
cantidad_producto_mas_vendido = ventas_por_producto.max()
print(f"Producto más vendido: {producto_mas_vendido}")
print("Unidades del producto más vendido: "f"{cantidad_producto_mas_vendido:,.0f}")

#Calculo gasto de marketing por canal 
marketing_por_canal = (marketing.groupby("canal")["gasto"].sum().sort_values(ascending=False))
print("\nGasto de marketing por canal:")
display(marketing_por_canal.to_frame(name="gasto_total"))


Ticket promedio: $385.85
Cantidad promedio de productos por orden: 1.50
Producto más vendido: Vacuum-Pro-Black
Unidades del producto más vendido: 6,284

Gasto de marketing por canal:


,gasto_total
canal,
social,976818.37
organic,972650.96
paid_search,922374.20


- El ticket promedio por pedido fue de 385.85.
- Cada pedido incluyó, en promedio, 1.50 unidades de productos.
- El producto más vendido fue Vacuum-Pro-Black, con 6,284 unidades vendidas.
- El canal con mayor inversión fue social, con 976,818.37, seguido de organic con 972,650.96 y paid_search con 922,374.20.

---

## 🔹 Paso 3: Entender dónde se pierden los usuarios (funnel de conversión)

**🎯 Objetivo:** Analizar el comportamiento de los usuarios para identificar en qué etapa del proceso se pierden.


⚙️**Conexión a la base de datos**:  
Se ejecuta la línea de configuración para conectar con la base de datos y aplicar consultas SQL en la tabla **events**.

---

**📊 Parte 1: Construcción del funnel**
- ¿Cuántos usuarios llegan a cada etapa del funnel?  
- Se calcula el número de usuarios únicos por `nombre_evento`  
- Se ordenan los eventos según el flujo del usuario  

---

**📉 Parte 2: Análisis de conversión**
- Se calcula la tasa de conversión entre cada paso del funnel  
- Se identifica en qué etapa se pierde la mayor cantidad de usuarios  
- ¿Cuál es la tasa de conversión final?
---

In [31]:
import pandas as pd
from sqlalchemy import create_engine

# ======================
# Conexión (NO modificar)
# ======================
db_config = {
    'user': 'practicum_student',
    'pwd': 'QnmDH8Sc2TQLvy2G3Vvh7',
    'host': 'yp-trainers-practicum.cluster-czs0gxyx2d8w.us-east-1.rds.amazonaws.com',
    'port': 5432,
    'db': 'data-analyst-production-db-en'
}

connection_string = 'postgresql://{}:{}@{}:{}/{}'.format(
    db_config['user'],
    db_config['pwd'],
    db_config['host'],
    db_config['port'],
    db_config['db']
)

engine = create_engine(connection_string, connect_args={'sslmode':'require'})

In [32]:
# Explorar tabla events
# =========================
query_events = '''
SELECT *
FROM events;
'''
events = pd.read_sql(query_events, con=engine)
events.head()

,id_usuario,id_sesion,nombre_evento,timestamp_evento,pais,dispositivo,fuente_referencia,categoria_producto
0,user_6772,6a97f2af-32ae-4186-8c92-04025be1a27b,first_visit,2025-05-17,Colombia,desktop,organic,Moda
1,user_5883,369b767c-1c33-4b2f-a652-c7c0ef92cfc9,add_to_cart,2025-02-23,Mexico,mobile,social,Hogar
2,user_5946,60039041-e78b-474c-87b3-c0b7e9c30708,add_payment_info,2025-05-15,Colombia,desktop,social,Electronica
3,user_827,18252a64-f389-4ef7-9e58-dadad4a3491e,purchase,2025-03-31,Mexico,mobile,social,Moda
4,user_2361,221b364e-cdc5-4668-b698-18d5ba849a67,first_visit,2025-01-22,Argentina,desktop,paid_search,Electronica


In [33]:
# PARTE 1: Totales del funnel
# ======================

query_totals = '''
SELECT
    nombre_evento,
    COUNT(DISTINCT id_usuario) AS usuarios_unicos
FROM events
GROUP BY nombre_evento
ORDER BY
    CASE nombre_evento
        WHEN 'first_visit' THEN 1
        WHEN 'select_item' THEN 2
        WHEN 'add_to_cart' THEN 3
        WHEN 'begin_checkout' THEN 4
        WHEN 'add_payment_info' THEN 5
        WHEN 'purchase' THEN 6
        ELSE 7
    END;
'''

totals = pd.read_sql(query_totals, con=engine)
totals

,nombre_evento,usuarios_unicos
0,first_visit,7796
1,select_item,7582
2,add_to_cart,7634
3,begin_checkout,7208
4,add_payment_info,6250
5,purchase,6240


In [34]:
# PARTE 2: Conversiones
# ======================

query_conversion = '''
WITH funnel AS (
    SELECT
        nombre_evento,
        COUNT(DISTINCT id_usuario) AS usuarios_unicos,

        CASE nombre_evento
            WHEN 'first_visit' THEN 1
            WHEN 'select_item' THEN 2
            WHEN 'add_to_cart' THEN 3
            WHEN 'begin_checkout' THEN 4
            WHEN 'add_payment_info' THEN 5
            WHEN 'purchase' THEN 6
            ELSE 7
        END AS orden_etapa

    FROM events
    GROUP BY nombre_evento
),

funnel_ordenado AS (
    SELECT
        nombre_evento,
        usuarios_unicos,
        orden_etapa,

        LAG(usuarios_unicos) OVER (
            ORDER BY orden_etapa
        ) AS usuarios_etapa_anterior,

        FIRST_VALUE(usuarios_unicos) OVER (
            ORDER BY orden_etapa
        ) AS usuarios_inicio

    FROM funnel
)

SELECT
    nombre_evento AS etapa,
    usuarios_unicos,

    usuarios_etapa_anterior - usuarios_unicos
        AS usuarios_perdidos,

    ROUND(
        usuarios_unicos::decimal
        / NULLIF(usuarios_etapa_anterior, 0) * 100,
        2
    ) AS tasa_conversion_paso

FROM funnel_ordenado
ORDER BY orden_etapa;
'''

conversion = pd.read_sql(query_conversion, con=engine)
conversion

,etapa,usuarios_unicos,usuarios_perdidos,tasa_conversion_paso
0,first_visit,7796,NaN,NaN
1,select_item,7582,214.0,97.26
2,add_to_cart,7634,-52.0,100.69
3,begin_checkout,7208,426.0,94.42
4,add_payment_info,6250,958.0,86.71
5,purchase,6240,10.0,99.84


Análisis del funnel de conversión

- El funnel comenzó con 7,796 usuarios en first_visit y finalizó con 6,240 usuarios en purchase. Esto representa una tasa de conversión final del 80.04%.

- La mayor pérdida se produjo entre begin_checkout y add_payment_info: la cantidad de usuarios disminuyó de 7,208 a 6,250, lo que representa una pérdida de 958 usuarios. La conversión entre estas etapas fue del 86.71%, por lo que el proceso de incorporación de la información de pago constituye el principal punto de abandono.

- Entre select_item y add_to_cart se observaron 52 usuarios adicionales, produciendo una conversión del 100.69%. Esto indica que el conteo agrupa usuarios únicos por evento de manera independiente y no garantiza que cada usuario haya completado todos los pasos anteriores. Algunos usuarios pudieron acceder directamente al carrito o no tener registrado el evento select_item.

- En la última transición, de add_payment_info a purchase, la conversión fue del 99.84%, lo que indica que casi todos los usuarios que ingresaron sus datos de pago terminaron la compra.

---

## 🔹 Paso 4: Evaluar si los usuarios regresan (retención por cohortes)

**🎯 Objetivo:** Analizar la retención de usuarios para entender si regresan después de registrarse.

**Tablas**

- `users` 
- `user_activity` 

---
1. Se identifica la cohorte de cada usuario según el **mes de registro**.


2. Se calcula la retención semanal: cuántos usuarios **se mantienen activos** en cada semana desde su registro.
   - `retenido_w1`: usuarios activos en la semana 1  
   - `retenido_w2`: usuarios activos en la semana 2  
   - `retenido_w3`: usuarios activos en la semana 3  


3. Se calcula el porcentaje de retención para cada semana, dividiendo los usuarios retenidos entre los clientes iniciales de la cohorte:  
   - `semana_1`: porcentaje de usuarios retenidos en la semana 1  
   - `semana_2`: porcentaje de usuarios retenidos en la semana 2  
   - `semana_3`: porcentaje de usuarios retenidos en la semana 3  

Se revisa que la columna de fecha esté en formato correcto (`DATE`).  
Se realiza la conversión usando: `CAST(fecha_registro AS DATE)`

In [35]:
# Explorar tabla users
# =========================
query_users = '''
SELECT *
FROM users;
'''
users = pd.read_sql(query_users, con=engine)
users.head(3)

,id_usuario,fecha_registro,país,dispositivo,tipo_plan
0,user_0,2025-01-29,Mexico,mobile,free
1,user_1,2025-01-07,Mexico,mobile,free
2,user_2,2025-03-12,Argentina,mobile,free


In [36]:
# Explorar tabla user_activity
# =========================
query_user_activity = '''
SELECT *
FROM user_activity;
'''
user_activity = pd.read_sql(query_user_activity, con=engine)
user_activity.head(3)

,id_usuario,fecha_actividad,dias_despues_registro,activo
0,user_0,2025-02-05,7,0
1,user_0,2025-02-12,14,1
2,user_0,2025-02-19,21,1


In [37]:
# Retención por cohortes
# ======================

query_cohort_retention_final = '''
WITH actividad_usuarios AS (
    SELECT
        u.id_usuario,
        DATE_TRUNC(
            'month',
            CAST(u.fecha_registro AS DATE)
        )::date AS cohorte,
        CAST(ua.fecha_actividad AS DATE) AS fecha_actividad,
        ua.dias_despues_registro,
        ua.activo
    FROM users AS u
    LEFT JOIN user_activity AS ua
        ON u.id_usuario = ua.id_usuario
),

retencion AS (
    SELECT
        cohorte,
        COUNT(DISTINCT id_usuario) AS clientes_iniciales,

        COUNT(DISTINCT CASE
            WHEN dias_despues_registro = 7
                 AND activo = 1
            THEN id_usuario
        END) AS retenido_w1,

        COUNT(DISTINCT CASE
            WHEN dias_despues_registro = 14
                 AND activo = 1
            THEN id_usuario
        END) AS retenido_w2,

        COUNT(DISTINCT CASE
            WHEN dias_despues_registro = 21
                 AND activo = 1
            THEN id_usuario
        END) AS retenido_w3

    FROM actividad_usuarios
    GROUP BY cohorte
)

SELECT
    cohorte,
    clientes_iniciales,
    retenido_w1,
    retenido_w2,
    retenido_w3,

    ROUND(
        retenido_w1::decimal /
        NULLIF(clientes_iniciales, 0) * 100,
        2
    ) AS semana_1,

    ROUND(
        retenido_w2::decimal /
        NULLIF(clientes_iniciales, 0) * 100,
        2
    ) AS semana_2,

    ROUND(
        retenido_w3::decimal /
        NULLIF(clientes_iniciales, 0) * 100,
        2
    ) AS semana_3

FROM retencion
ORDER BY cohorte;
'''

# Ejecutar la consulta
cohorte_final = pd.read_sql(query_cohort_retention_final, con=engine)
cohorte_final

,cohorte,clientes_iniciales,retenido_w1,retenido_w2,retenido_w3,semana_1,semana_2,semana_3
0,2025-01-01,1627,697,668,656,42.84,41.06,40.32
1,2025-02-01,1444,611,609,635,42.31,42.17,43.98
2,2025-03-01,1636,677,705,690,41.38,43.09,42.18
3,2025-04-01,1606,680,697,663,42.34,43.40,41.28
4,2025-05-01,1687,695,676,706,41.20,40.07,41.85


La retención semanal se mantiene relativamente estable entre las cohortes analizadas, con valores cercanos al 42% durante las primeras tres semanas posteriores al registro.

- La retención promedio fue aproximadamente de 42.01% en la semana 1, 41.96% en la semana 2 y 41.92% en la semana 3.
- La cohorte de enero de 2025 presentó la retención más alta en la semana 1, con 42.84%.
- La cohorte de abril de 2025 alcanzó la mayor retención en la semana 2, con 43.40%.
- La cohorte de febrero de 2025 registró la mayor retención en la semana 3, con 43.98%.

---

## 🔹 Paso 5: Validar si los cambios generan impacto (test estadístico)

🎯 **Objetivo:** Evaluar si la modificación en la UI del checkout impacta la **tasa de conversión de compra**.

---

1. **Analizar el dataset** `experiment_checkout_ui.csv` para identificar la métrica principal **conversion**.
   - La métrica **conversion** es 1 si el usuario completó la compra, 0 si no.    
2. **Plantear la hipótesis estadística**     
3. **Aplicar el test estadístico adecuado** 
4. **Interpretar el resultado**  

---
Hipótesis estadística
   - **H₀ (Hipótesis nula):** No existe diferencia en la tasa de conversión de compra entre el grupo con la interfaz original y el grupo con la interfaz modificada. 
   - **H₁ (Hipótesis alternativa):** Existe diferencia en la tasa de conversión de compra entre el grupo con la interfaz original y el grupo con la interfaz modificada. 
   
**Test estadístico:** Prueba Z  
**Nivel de significancia alpha:** 0.05

In [38]:
# Cargar y analizar el dataset experimento
experiment = pd.read_csv('https://practicum-content.s3.amazonaws.com/datasets/experiment_checkout_ui.csv')
display(experiment.head())

,id_usuario,variante,convirtio,dispositivo,pais,duracion_sesion,timestamp
0,exp_user_0,tratamiento,0,mobile,Argentina,114.41,2025-03-28
1,exp_user_1,tratamiento,0,desktop,Mexico,170.03,2025-01-15
2,exp_user_2,control,1,mobile,Colombia,140.21,2025-03-18
3,exp_user_3,tratamiento,0,mobile,Colombia,151.45,2025-06-03
4,exp_user_4,tratamiento,0,desktop,Mexico,299.96,2025-01-12


In [39]:
from statsmodels.stats.proportion import proportions_ztest

# Contar número de conversiones y total de usuarios
conversiones = experiment.groupby("variante")["convirtio"].sum()
totales = experiment.groupby("variante")["convirtio"].count()

# Pasar los valores a formato lista
exitos = [
    conversiones["control"],
    conversiones["tratamiento"]
]

observaciones = [
    totales["control"],
    totales["tratamiento"]
]

# Mostrar las tasas de conversión
tasa_control = conversiones["control"] / totales["control"]
tasa_tratamiento = (
    conversiones["tratamiento"] /
    totales["tratamiento"]
)

print(f"Tasa de conversión del control: {tasa_control:.2%}")
print(
    "Tasa de conversión del tratamiento: "
    f"{tasa_tratamiento:.2%}"
)

# Aplicar prueba Z de dos proporciones
z_stat, p_value = proportions_ztest(
    exitos,
    observaciones
)

print(f"\nEstadístico z: {z_stat:.4f}")
print(f"Valor p: {p_value:.4f}")

# Interpretar los resultados
alpha = 0.05

if p_value < alpha:
    print(
        "Rechazamos la hipótesis nula: existe evidencia "
        "estadísticamente significativa de una diferencia "
        "en la tasa de conversión."
    )
else:
    print(
        "No rechazamos la hipótesis nula: no existe evidencia "
        "estadística suficiente de una diferencia en la tasa "
        "de conversión."
    )

Tasa de conversión del control: 15.69%
Tasa de conversión del tratamiento: 16.29%

Estadístico z: -0.8133
Valor p: 0.4161
No rechazamos la hipótesis nula: no existe evidencia estadística suficiente de una diferencia en la tasa de conversión.


In [40]:
# Obtener tasa o porcentaje de conversión
tasa_control = exitos[0] / observaciones[0]
tasa_tratamiento = exitos[1] / observaciones[1]

print(
    f"Tasa de conversión del grupo control: "
    f"{tasa_control:.2%}"
)

print(
    f"Tasa de conversión del grupo tratamiento: "
    f"{tasa_tratamiento:.2%}"
)

# Interpretar la dirección de los resultados
if tasa_control > tasa_tratamiento:
    diferencia = tasa_control - tasa_tratamiento

    print(
        f"\nEl grupo control tiene una mayor tasa de conversión. "
        f"La diferencia es de {diferencia:.2%}."
    )

elif tasa_tratamiento > tasa_control:
    diferencia = tasa_tratamiento - tasa_control

    print(
        f"\nEl grupo tratamiento tiene una mayor tasa de conversión. "
        f"La diferencia es de {diferencia:.2%}."
    )

else:
    print(
        "\nAmbos grupos tienen la misma tasa de conversión."
    )

Tasa de conversión del grupo control: 15.69%
Tasa de conversión del grupo tratamiento: 16.29%

El grupo tratamiento tiene una mayor tasa de conversión. La diferencia es de 0.60%.


- El grupo control obtuvo una tasa de conversión del 15.69%, mientras que el grupo tratamiento alcanzó una tasa del 16.29%. Descriptivamente, la interfaz modificada presentó una conversión 0.60 puntos porcentuales mayor.

- Sin embargo, la prueba Z de dos proporciones produjo un estadístico de −0.8133 y un valor p de 0.4161. Debido a que el valor p es mayor que el nivel de significancia de 0.05, no se rechaza la hipótesis nula.

- Por lo tanto, no existe evidencia estadística suficiente para afirmar que la modificación de la interfaz del checkout haya generado un cambio real en la tasa de conversión. Aunque el tratamiento mostró una conversión ligeramente superior, la diferencia observada no es estadísticamente significativa.

---

## 🔹 Paso 6: Comunicar los resultados (Dashboard en BI)

🎯 **Objetivo**:  
Crear un dashboard que muestre de manera clara y visual los resultados del análisis de ventas, costos, marketing y conversión. 

Se usarán los CSVs limpios del Paso 1:

- `orders_clean.csv`  
- `catalog_clean.csv`  
- `marketing_clean.csv`

---

1️⃣ Preparación de los datos
1. Cargar los CSVs en Power BI o Tableau.
2. Revisar relaciones:
   - `orders.nombre_producto` → `catalog.nombre_producto`
   - `orders.fecha_pedido` → tabla de fechas (crear calendario para análisis temporal)
   - `orders.fecha_pedido` → `dim_fecha.date`
3. Crear columnas calculadas necesarias
4. Crear tabla de fechas para poder calcular comparaciones YTD, YoY o períodos anteriores (`Previous Year`, `Previous Month`).

---

2️⃣ Dashboard 1: Overview Ejecutivo
**KPIs principales a mostrar:**
- Revenue total
- Profit total
- Gasto total en marketing
- Ticket promedio
- Cantidad promedio de productos por orden

**Visualizaciones sugeridas:**
- Tarjetas KPI para revenue, profit y gasto marketing
- Gráfico de líneas: evolución mensual de revenue o profit
- Gráfico de líneas YTD
- Gráfico de barras: revenue y profit por producto o categoría

---

 3️⃣ Dashboard 2: Detalle / Drill-through  
**Objetivo:** Permitir explorar los datos desde el KPI general hasta cada orden o producto.



**Visualizaciones sugeridas:**
- Tabla detallada de órdenes con:
  - producto, cantidad, revenue, cost, profit
  - color condicional (profit negativo en rojo, positivo en verde)
- Gráfico de barras por producto con medida `cantidad vendida`
- Drill-through: seleccionar un producto y ver todos los pedidos relacionados
- Filtros por fecha, categoría de producto, etc

---

## 🚀 Entrega Final

Comparte el acceso a tu Dashboard para revisión.   
Puedes entregar el Dashboard utilizando **Power BI o Tableau**.

Incluye **uno de los siguientes**:

- 🔗 Link público del dashboard publicado en **Power BI Service o Tableau Public / Tableau Cloud**
- 🔗 Link de **Google Drive o OneDrive** con el archivo del proyecto (`.pbix`) y los 3 csvs limpios.


### 📎 Enlace del Dashboard

In [41]:
# https://drive.google.com/drive/folders/1cqYpfbUEzRA-98hALonruDckwP5r6lpK?usp=sharing